votor-messages/src/consensus_message.rs

```diff
@@ -143,36 +196,96 @@ pub struct CertificateMessage {
     pub bitmap: Vec<u8>,
 }
 
+#[cfg_attr(
+    feature = "frozen-abi",
+    derive(AbiExample, AbiEnumVisitor),
+    frozen_abi(digest = "CwKtX5nWfGbQZSBh1YTr3NABwThE4zFTxzQL9K5xkqYW")
+)]
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 #[allow(clippy::large_enum_variant)]
 /// BLS message data in Alpenglow
-pub enum ConsensusMessage {
+pub enum Message {
     /// Vote message, with the vote and the rank of the validator.
     Vote(VoteMessage),
     /// Certificate message
     Certificate(CertificateMessage),
 }
 
-impl ConsensusMessage {
-    /// Create a new vote message
-    pub fn new_vote(vote: Vote, signature: BLSSignature, rank: u16) -> Self {
-        Self::Vote(VoteMessage {
-            vote,
-            signature,
-            rank,
-        })
+#[cfg(test)]
+mod test {
+    use {super::*, bincode};
+
+    #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+    struct ConsensusMessageUnknown {
+        version_major: u8,
+        version_minor: u8,
+        message: MessageUnknown,
     }
 
-    /// Create a new certificate message
-    pub fn new_certificate(
-        certificate: Certificate,
-        bitmap: Vec<u8>,
-        signature: BLSSignature,
-    ) -> Self {
-        Self::Certificate(CertificateMessage {
-            certificate,
-            signature,
-            bitmap,
-        })
+    #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+    pub enum MessageUnknown {
+        /// Vote message, with the vote and the rank of the validator.
```

Comment:
I am not convinced the test is faithful since Unknown message is the same one as known one, i.e. it will pass deserialize no matter what. To properly test, we need to include dynamic length fields such that "old" deserialize impl would get confused when faced with "new" message.